week2_shoppinglist/index.js

```diff
@@ -0,0 +1,384 @@
+    image: "./images/note.jpg",
+    name: "잔망루피 페이스 스프링 노트",
+    price: 4000,
+    liked: false,
+  },
+  {
+    id: 2,
+    category: "stationery",
+    image: "./images/bag.jpg",
+    name: "잔망루피 쇼핑백 (2종/택1)",
+    price: 1500,
+    liked: false,
+  },
+  {
+    id: 3,
+    category: "stationery",
+    image: "./images/sticker.jpg",
+    name: "잔망루피 스티커",
+    price: 1500,
+    liked: false,
+  },
+  {
+    id: 4,
+    category: "fashion",
+    image: "./images/doll.jpg",
+    name: "잔망루피 파자마 봉제 인형",
+    price: 29000,
+    liked: false,
+  },
+  {
+    id: 5,
+    category: "fashion",
+    image: "./images/passport.jpg",
+    name: "잔망루피 여권 케이스",
+    price: 5900,
+    liked: false,
+  },
+  {
+    id: 6,
+    category: "fashion",
+    image: "./images/flower.jpg",
+    name: "잔망루피 꽃다발 봉제 인형",
+    price: 34000,
+    liked: false,
+  },
+  {
+    id: 7,
+    category: "digital",
+    image: "./images/mouse.jpg",
+    name: "잔망루피 마우스패드",
+    price: 4000,
+    liked: false,
+  },
+  {
+    id: 8,
+    category: "digital",
+    image: "./images/usb.jpg",
+    name: "로이체 잔망루피 5in1 멀티 USB 허브",
+    price: 36900,
+    liked: false,
+  },
+  {
+    id: 9,
+    category: "digital",
+    image: "./images/mic.jpg",
+    name: "로이체 잔망루피 블루투스 마이크 스피커",
+    price: 39900,
+    liked: false,
+  },
+];
+
+// 상품 리스트 렌더링
+function renderItems(filterCategory = "all") {
+  const filteredItems = SHOPPING_LIST.filter((item) => {
+    return filterCategory === "all" || item.category === filterCategory; // 상품 카테고리 필터링
+  });
+
+  const itemsContainer = document.querySelector(".items"); // 상품 리스트를 렌더링할 class 가져오기
+  itemsContainer.innerHTML = ""; // 기존 상품 목록 비우기
+
+  // 상품 목록 추가
+  filteredItems.forEach((item) => {
+    const article = document.createElement("article"); // 각 상품에 대한 article 생성
+    article.className = "item";
+    article.onclick = function () {
+      addCart(item);
+    };
+
+    // 이미지
+    const img = document.createElement("img");
+    img.src = item.image;
+    img.alt = "상품 이미지";
+    img.className = "item-image";
+
+    // 좋아요
+    const likeDiv = document.createElement("div");
+    likeDiv.className = "like";
+
+    // 상품명
+    const nameP = document.createElement("p");
+    nameP.className = "name";
+    nameP.textContent = item.name;
+
+    // 가격
+    const priceP = document.createElement("p");
+    priceP.className = "price";
+    priceP.textContent =
+      item.price.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",") +
+      "원"; // 천단위 표시
+
+    // 요소 추가
+    article.appendChild(img);
+    article.appendChild(likeDiv);
+    article.appendChild(nameP);
+    article.appendChild(priceP);
+    itemsContainer.appendChild(article);
+  });
+}
+
+//장바구니 목록 추가
+function addCart(item) {
+  let cartList = JSON.parse(localStorage.getItem("cartList")) || []; // cartList가 배열이 아닌 경우, 빈 배열로 초기화
+  cartList.push(item); // 새 상품 추가
+  localStorage.setItem("cartList", JSON.stringify(cartList)); // 변경된 장바구니 목록을 로컬 스토리지에 저장
```

Comment:
여기서 push 함수보다는 이번 공유과제에서 배웠던 배열의 불변성의 법칙을 지키면서 추가하는 건 어떨까~?

```suggestion
  const newCartList = [...cartList, item];  // 새 상품 추가
  localStorage.setItem("cartList", JSON.stringify(newCartList)); // 변경된 장바구니 목록을 로컬 스토리지에 저장
```
이렇게 할 수 있을 것 같애!

Reply:
오🫢직접 코드까지 수정해주시다니!! 감사합니다!! 배열의 불변성을 항상 안지켜왔던 것 같아서 반성합니다...
이번 공유 과제를 한번 적용해보도록 하겠습니다!

buyList까지 적용했습니다!ㅎㅎ